### ISSUE TITLE
TestFx and QuarkusTest seem incompatible

### ISSUE THREAD

**Issue by jjfraney** [TestFX/TestFX#796]
## Expected Behavior

QuarkusTest and testFx should be compatible.

## Actual Behavior

They seem mutually exclusive.

1) QuarkusTest only: Scene not loaded. @Start does not run.

2) TestFx only: Scene not loaded. FXMLloader is not injected - no CDI container.

3) Both enabled: Scene not loaded. @Start runs but FXMLloader field is null; Start runs before the container or Start and Test are on different proxies/instances (?).

To get the fastest possible support, create a [minimal example](https://stackoverflow.com/help/mcve)
and upload it to GitHub.

For a runnable example, please clone this small project, and check out the branch named testfx
   https://github.com/jjfraney/quarkus-javafx-example.git

## Specifications

  - Version: testfx-junit5:4.0.17

  - Platform:  JDK 17

**Comment by Paranoja** [comment]
In case someone else stumbles upon this.

TL;DR: when one uses `@QuarkusTest` there are **two** instances of test class created -- one by JUnit5 and one by Quarkus. To use TestFX together with Quarkus and JUnit5 one shall ensure that Quarkus is controlling the lifecycle of TestFX by in `QuarkusTest*Callback`s.

Quarkus-JUnit5 cooperation works, if I understand it correcty, in following way: JUnit5 first test class with `@QuarkusTest` extension kicks initialization of Quarks framework off. Quarkus then builds its CDI context (starts istelf) and registers itself in JUnit5 context. JUnit5 creates its instances of test classes to be able to, i.e., determine which test methods are parameterized, which are test factories etc. Quarkus, on the other hand, to support its test features, like `@Transactional` proxies, field injection (CDI), `@InjectMock`/`@InjectSpy`, generates proxies during **compilation** and uses separate classloader for that. See last warning in [Enrichment via QuarkusTest*Callback](https://quarkus.io/guides/getting-started-testing#enrichment-via-quarkustestcallback) and [Class Loading Reference](https://quarkus.io/guides/class-loading-reference) for some extra details.

Hence at least two instances of test class are created: one, instantiated by JUnit5, and one -- by Quarkus. That also depend on [`@TestInstance`](https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-instance-lifecycle) setting.

I suggest using QuarkusTest*Callbacks to [integrate with TestFX](https://github.com/TestFX/TestFX/wiki/Stages-and-Applications#general-approach-for-integration-and-unit-testing) instead of `@ExtendWith(ApplicationExtension.class)` like this:

```
import io.quarkus.test.junit.callback.QuarkusTestBeforeEachCallback;

@QuarkusTest
class FooTest implements QuarkusTestBeforeEachCallback {

  @Override
  public void beforeEach(QuarkusTestMethodContext context) {
      FxToolkit.registerPrimaryStage();
      FxToolkit.setupApplication(FooApplication.class);
  }

  @Test
  void foo() {
    FxAssert.verifyThat(...);
  }
}

```

See related issue quarkusio/quarkus/issues/34330.
